packages/smart-wallet/src/smartWallet.js

```diff
@@ -972,13 +972,14 @@ export const prepareSmartWallet = (baggage, shared) => {
 
             const invitation = invitationFromSpec(offerSpec.invitationSpec);
 
-            const [paymentKeywordRecord, invitationAmount] = await Promise.all([
+            const invitationAmount =
+              await E(invitationIssuer).getAmountOf(invitation);
+
+            const paymentKeywordRecord =
```

Comment:
This is mutative, so no longer simply "prepare". It belongs under "2. Begin executing offer". Consider also naming the object like `withdrawnPayments` to it's more clear how it relates to the lifecycle of the request.

Reply:
done.